odoo/orm/fields_numeric.py

```diff
@@ -283,8 +283,8 @@ def _filter_not_equal(self, records: BaseModel, cache_value: typing.Any) -> Base
                 records._ids, records.sudo().with_context(prefetch_fields=False)
             )
             if not (
-                (value := field_cache.get(record_id))
+                (value := field_cache.get(record_id)) is not None
```

Comment:
This change is not needed, since falsy values are rounded.

Reply:
if you `cache_value` is `0.0` and `value` is `0.0`
`_filter_not_equal` shouldn't return the record